fennel.lua

```diff
@@ -1221,10 +1245,23 @@ SPECIALS['fn'] = function(ast, scope, parent)
             assertCompile(false, 'expected symbol for function parameter', ast)
         end
     end
+    if type(ast[index + 1]) == 'string' and index + 1 < #ast then
+      index = index + 1
+      docstring = ast[index]
+      if rootOptions.useMetadata then
+        docstring = docstring -- account for newlines and nested [[ ]] multiline strings
+                :gsub('\n', '\\n')
+                :gsub('[[](=*)[[]', '[=%1['):gsub('[]](=*)[]]', ']=%1]')
```

Comment:
I think the `[[this style]]` of strings is valid for Lua but multiline Fennel strings just use normal quotes?

Reply:
That's there to account for when emitting the lua that sets the docstring on line 1279
```lua
local docSetter = string.format('require("fennel").metadata:set(%s, "docstring", [[%s]])',
```

since, if someone were to put `[[ ]]` in the docstring without the above escaping of `[[ ]]`, it'd break. Though, upon reflection, `:gsub('\n', '\\n')` may be unnecessary. I'll add some tests for multiline docstrings and see. I'll also throw in some tests for docstrings with `[[ ]]` just to futureproof it.

I suppose I could just pass the docstring into `compile` or `compile1` to just let the compiler's existing string-juggling handle all that?

Reply:
Actually, leaving in `:gsub('\n', '\\n')`, removing the `[[ ]]` handling and changing line 1279 to
```lua
local docSetter = string.format('require("fennel").metadata:set(%s, "docstring", "%s")',
```
works! It makes the docstring less readable in the compiled lua than using `[[ ]]` but also means emitting fewer lines of code when there are multiline docstrings so it's kind of an even trade.

Reply:
Oh, I see. I misunderstood the purpose of that snippet, so go ahead and ignore my comment.

Reply:
Agreed - unless I’m mistaken, `[[strings like these]]` aren’t currently valid in Fennel, and it’s a syntax that collides with Fennel’s sequential-table literals.